faults/common.go

```diff
@@ -0,0 +1,124 @@
+// Package faults provides common headers and client-side fault injection functionality.
+package faults
+
+import (
+	"fmt"
+	"log/slog"
+	"math/rand/v2"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type GetHeaderFn func(key string) string
+type ResumeFn func() (interface{}, error)
+type ResponseFn func(code int, message string) (interface{}, error)
+type SleepFn func(d time.Duration)
+
+// Object to ensure a random number is only generated at most 1 time.
+type randSingleton struct {
+	randInt *int
+}
+
+func (r randSingleton) getRandInt() int {
+	if r.randInt == nil {
+		*r.randInt = rand.IntN(100)
+	}
+	return *r.randInt
+}
+
+func isSelected(percentageHeader string, GetHeaderFn func(key string) string, singleRand randSingleton) (bool, string) {
+	percentageStr := GetHeaderFn(percentageHeader)
+	if percentageStr == "" {
+		return true, ""
+	}
+	percentage, err := strconv.Atoi(percentageStr)
+	if err != nil {
+		return false, fmt.Sprintf("provided percentage \"%s\" is not a valid integer", percentageStr)
```

Comment:
1. use `%q` over `\"%s\"`
2. also wrap the original error?
```suggestion
		return false, fmt.Sprintf("provided percentage %q is not a valid integer: %w", percentageStr, err)
```

Reply:
oh you are returning a string instead of error for the error. I don't really see a reason for that, please return an error instead.

Reply:
Updated!